cli/utils/calculate-rewards.ts

```diff
@@ -91,19 +95,26 @@ export function processAttestationCompletion(
     issuers[account].push(issuer)
     state.attestationCompletions[account] += 1
   }
+}
 
-  if (state.startedBlockBalanceTracking && state.attestationCompletions[account] === 3) {
-    // Just completed
-    state.balancesByBlock[account] = {
-      balances: [state.balances[account] || new BigNumber(0)],
-      blocks: [event.blockNumber],
-    }
+export function processAccountWalletAddressSet(walletAssociations: WalletAssociations, event: EventLog) {
+  const account = event.returnValues.account
+  const wallet = event.returnValues.walletAddress
+  if (!walletAssociations[wallet]) walletAssociations[wallet] = [wallet] // the wallet itself may be verified
+  if (!walletAssociations[wallet].includes(account)) walletAssociations[wallet].push(account)
+}
+
+export function accountIsVerified(state: RewardsCalculationState, account: string): boolean {
+  let walletAssociations = state.walletAssociations
+  let associatedAddresses = walletAssociations[account] ? walletAssociations[account] : [account]
+  for (let i in associatedAddresses) {
```

Comment:
nit: slightly easier to read
`return associatedAddresses.some(address => state.attestationCompletions[address] >= 3)`

Reply:
oh I love this!